crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::crypto::PublicKey;
+use crate::crypto::Secp256r1SuiSignature;
+use crate::crypto::SuiSignatureInner;
+use crate::signature_verification::VerifiedDigestCache;
+use crate::{
+    base_types::{EpochId, SuiAddress},
+    crypto::{DefaultHash, Signature, SignatureScheme, SuiSignature},
+    digests::ZKLoginInputsDigest,
+    error::{SuiError, SuiResult},
+    signature::{AuthenticatorTrait, VerifyParams},
+};
+use fastcrypto::hash::{HashFunction, Sha256};
+use fastcrypto::rsa::{Base64UrlUnpadded, Encoding};
+use fastcrypto::secp256r1::{Secp256r1PublicKey, Secp256r1Signature};
+use fastcrypto::traits::VerifyingKey;
+use fastcrypto::{error::FastCryptoError, traits::ToFromBytes};
+use once_cell::sync::OnceCell;
+use passkey::types::webauthn::{ClientDataType, CollectedClientData};
+use schemars::JsonSchema;
+use serde::{Deserialize, Deserializer, Serialize};
+use shared_crypto::intent::Intent;
+use shared_crypto::intent::{IntentMessage, INTENT_PREFIX_LENGTH};
+use std::hash::Hash;
+use std::hash::Hasher;
+use std::sync::Arc;
+
+#[cfg(test)]
+#[path = "unit_tests/passkey_authenticator_test.rs"]
+mod passkey_authenticator_test;
+
+/// An passkey authenticator with parsed fields. See field defition below. Can be initialized from [struct RawPasskeyAuthenticator].
+#[derive(Debug, Clone, JsonSchema)]
+pub struct PasskeyAuthenticator {
+    /// `authenticatorData` is a bytearray that encodes
+    /// [Authenticator Data](https://www.w3.org/TR/webauthn-2/#sctn-authenticator-data)
+    /// structure returned by the authenticator attestation
+    /// response as is.
+    authenticator_data: Vec<u8>,
+
+    /// `clientDataJSON` contains a JSON-compatible
+    /// UTF-8 encoded string of the client data which
+    /// is passed to the authenticator by the client
+    /// during the authentication request (see [CollectedClientData](https://www.w3.org/TR/webauthn-2/#dictdef-collectedclientdata))
+    client_data_json: String,
+
+    /// Normalized r1 signature returned by passkey.
+    /// Initialized from `user_signature` in `RawPasskeyAuthenticator`.
+    #[serde(skip)]
```

Comment:
Do we still need `#[serde(skip)]`? Does `JsonSchema` use it?

Reply:
yeah JsonSchema recognizes serde attrs, but that's a good question on if we do need to include it or not

Reply:
this is a dumb question, how is this open json rpc generated spec being used?

Reply:
honestly, for crypto stuff i think it can be ignored for now since afaik we just serialize crypto sigs to a base64 encoding in jsonrpc. So this is all ignored I believe.

Reply:
i see, ill leave it as is then.